File: python/projekt_magazyn/sIG_zal.py
#!/usr/bin/python3
#Izabela Gorlinska grupa 1
#skrypt zaliczeniowy python

from Istnieje import istnieje
from sys import argv
from os.path import isfile
import csv
import re

class kolor:
	RED='\033[91m'
	GREEN='\033[92m'
	RESET='\033[0m'

def addProduct(spis):
	print("Dodajesz produkt do spisu - " +spis +" Podaj dane produktu." )
	nazwa=input("Nazwa: ")
	ilosc=input("Ilosc: ")
	i=0
	while i<3:
		if not ilosc.isnumeric():
			if i==0:			
				ilosc=input("To nie liczba sztuk! Podaj poprawnie: ")
			elif i==2:
				print(kolor.RED +"Podano nieprawidlowa wartosc! Nie udalo sie dodac produktu :(" +kolor.RESET)
				return
			else:
				ilosc=input("Sprobuj jeszcze raz: ")
			i+=1
		else:
			i=3
	cena=input("Cena: ")
	if not re.search('[0-9]+$',cena):
		cena=input("Podaj poprawna cene: ")
	cena=float(cena)
	suma=round(float(ilosc)*cena,3)
	with open(spis, 'a+', newline='') as write_file:
		csv_writer=csv.writer(write_file)
		csv_writer.writerow([nazwa, ilosc, cena, suma])
	print(kolor.GREEN +"Pomyslnie dodano produkt do spisu :)" +kolor.RESET)


def removeProduct(spis):
	print("Usuwasz produkt ze spisu - " +spis)
	showProducts(spis)
	decyzja='x'
	list=[]
	prod=input("Podaj nazwe produktu, ktory chcesz usunac: ")
	with open(spis) as csv_file:
		csv_reader = csv.reader(csv_file, delimiter=',')
		for row in csv_reader:
			if row[0]==prod:
				decyzja=input("Czy na pewno chcesz usunac " +prod +" ? (tak/nie)")
			else:
				list.append(row)
		if decyzja =='x':
			print("Nie ma produktu o takiej nazwie :(")
		elif decyzja =='tak':
			with open(spis, 'w', newline='') as write_file:
				csv_writer=csv.writer(write_file)
				csv_writer.writerows(list)
			print(kolor.GREEN +"Produkt zostal ususniety." +kolor.RESET)
		else:
			print("Ok. Nie bedziemy usuwac")
			return
	showProducts(spis)
					

def changeProduct(spis):
	print("Zmieniasz dane produktu ze spisu - " +spis)
	akcja=input("Wybierz co chcesz zmienic:\n 1)nazwa\n 2)ilosc\n 3)cena\nTwoj wybor: ")
	if akcja=='1':
		print("Zmieniasz nazwe.")
		prod=input("Podaj nazwe produktu, ktory chcesz edytowac: ")
		ex=istnieje(spis, prod, 0)
		if not ex==1:
			prod=input("Podaj nazwe produktu jeszcze raz: ")
			ex=istnieje(spis, prod, 1)
		if ex==1:
			list=[]
			with open(spis) as csv_file:
				csv_reader = csv.reader(csv_file, delimiter=',')
				for row in csv_reader:
					if row[0]==prod:
						proded=row
						nowa=input("Podaj nowa nazwe: ")
						proded[0]=nowa
						list.append(row)
					else:
						list.append(row)
				decyzja=input("Czy na pewno chcesz zmienic nazwe " +prod +" na " +nowa +" ? (tak/nie)")
				if decyzja =='tak':
					with open(spis, 'w', newline='') as write_file:
						csv_writer=csv.writer(write_file)
						csv_writer.writerows(list)
						print(kolor.GREEN +"Edycja zakonczona sukcesem :)" +kolor.RESET)
					print
				else:
					print(kolor.RED +"Edycja wstrzymana" +kolor.RESET)
					return

	
	elif akcja=='2':
		print("Zmieniasz ilosc.")
		prod=input("Podaj nazwe produktu, ktory chcesz edytowac: ")
		ex=istnieje(spis, prod, 0)
		if not ex==1:
			prod=input("Podaj nazwe produktu jeszcze raz: ")
			ex=istnieje(spis, prod, 1)
		if ex==1:
			list=[]
			with open(spis) as csv_file:
				csv_reader = csv.reader(csv_file, delimiter=',')
				for row in csv_reader:
					if row[0]==prod:
						proded=row
						nowa=input("Podaj nowa ilosc: ")
						if not nowa.isnumeric():
							nowa=input("Podaj poprawna ilosc: ")
							if not nowa.isnumeric():
								nowa=input("Podales niepoprawna ilosc. Sprobuj jeszcze raz: ")
								if not nowa.isnumeric():
									print("Wprowadzona ilosc nie jest poprawna. Przerwano zmienianie ilosci produktu")
									return
						proded[1]=nowa
						proded[2]=float(proded[2])
						proded[3]=round(float(proded[1])*proded[2],3)
						list.append(row)
					else:
						list.append(row)
				decyzja=input("Czy na pewno chcesz zmienic ilosc " +prod +" ? (tak/nie)")
				if decyzja =='tak':
					with open(spis, 'w', newline='') as write_file:
						csv_writer=csv.writer(write_file)
						csv_writer.writerows(list)
					print(kolor.GREEN +"Edycja zakonczona sukcesem :)" +kolor.RESET)
				else:
					print(kolor.RED +"Edycja wstrzymana" +kolor.RESET)
					return	
	elif akcja=='3':
		print("Zmieniasz cene.")
		prod=input("Podaj nazwe produktu, ktory chcesz edytowac: ")
		ex=istnieje(spis, prod, 0)
		if not ex==1:
			prod=input("Podaj nazwe produktu jeszcze raz: ")
			ex=istnieje(spis, prod, 1)
		if ex==1:
			list=[]
			with open(spis) as csv_file:
				csv_reader = csv.reader(csv_file, delimiter=',')
				for row in csv_reader:
					if row[0]==prod:
						proded=row
						print("Obecna cena: " +row[2])
						nowa=input("Podaj nowa cene: ")
						if not re.search('[0-9]+$',nowa):
							nowa=input("Podaj poprawna cene: ")
							if not re.search('[0-9]+$',nowa):
								nowa=input("Podales niepoprawna cene. Sprobuj jeszcze raz: ")
								if not re.search('[0-9]+$',nowa):
									print("Wprowadzona cena nie jest poprawna. Przerwano zmienianie ilosci produktu")
									return
						proded[2]=float(nowa)
						proded[3]=round(float(proded[1])*proded[2],3)
						list.append(row)
					else:
						list.append(row)
				decyzja=input("Czy na pewno chcesz zmienic cene " +prod +" ? (tak/nie)")
				if decyzja =='tak':
					with open(spis, 'w', newline='') as write_file:
						csv_writer=csv.writer(write_file)
						csv_writer.writerows(list)
					print(kolor.GREEN +"Edycja zakonczona sukcesem :)" +kolor.RESET)
				else:
					print(kolor.RED+"Edycja wstrzymana" +kolor.RESET)
					return
	else:
		print(kolor.RED +"Nie wybrano numeru akcji. Uruchom skrypt ponownie jesli chcesz edytowac" +kolor.RESET)



def showProducts(spis):
	print("\nWyswietlam spis - " +spis )
	with open(spis) as csv_file:
		csv_reader = csv.reader(csv_file, delimiter=',')
		#line_count = 0
		for row in csv_reader:
			print (row[0], row[1], row[2], row[3])
			#line_count += 1


def summary(spis):
	print("\nWyswietlam dane ogolne dla spisu - " +spis )
	with open(spis) as csv_file:
		csv_reader = csv.reader(csv_file, delimiter=',')
		lprod = 0
		lil=0
		lsum=0
		for row in csv_reader:
			lprod += 1
			lil+=int(row[1])
			lsum+=float(row[3])
	print("Liczba produktow w spisie: " +str(lprod) +"\nLacznie na magazynie jest: " +str(lil) +" sztuk wszystkich towarow\nWartosc wszystkich produktow wynosi: "+str(round(lsum,3)) +" zl")


def printFile(spis):
	plikz = open('Summary_py.txt','w')
	plikz.write("Podsumowanie spisu towarow z magazynu\n\nPRODUKTY\nNazwa produktu, ilosc, cena\n===========================\n")
	with open(spis) as csv_file:
		csv_reader = csv.reader(csv_file, delimiter=',')
		lprod = 0
		lil=0
		lsum=0
		for row in csv_reader:
			plikz.write("%s %s %.2f %.2f\n" % (row[0], row[1], float(row[2]), float(row[3])))
			lprod += 1
			lil+=int(row[1])
			lsum+=float(row[3])
	plikz.write("\nPODSUMOWANIE\nLiczba produktow w spisie: %i\nLacznie na magazynie jest: %i sztuk wszystkich towarow\nWartosc wszystkich produktow wynosi: %.2f zl" % (lprod, lil, lsum))
	plikz.close()


#koniec def
h=0
f=0

if len(argv)<=1:
	print (kolor.RED +"Nie podano pliku do analizy.\nJesli chcesz dowiedziec sie co ja robie wybierz opcje '-help' :)" +kolor.RESET)
	exit(1)
else:
	for pom in range (1, len(argv)):
		if argv[pom] == '-h' or argv[pom] == '-help':
				if h==0:
					print ("Wywolales skrypt z opcja help.")
					print("Autor: Izabela Gorlinska\n")
					print("Ten program jako pierwszy argument przyjmuje plik z danymi (plik csv) - spisem stanu towarow na magazynie i na nim operuje.\nProgram akceptuje tylko 1 plik i na nim pracuje.\nPo wywolaniu konkretnej opcji wykonuje odpowiednia dla niej akcje.\nSkrypt mozna uruchomic z 1 lub wieloma opcjami na raz.")
					print("\nSkrypt wywolany bez parametrow wypisuje wartosci wszystkich towarow w magazynie oraz liczbę towarów i ich laczna ilosc.\nMozliwe opcje wywolania skryptu:")
					print(" -add     dodanie produktu do magazynu,")
					print(" -rem     usuniecie produktu z magazynu,")
					print(" -change  zmiana nazwy,ilosci lub ceny produktu,")
					print(" -show    wypisanie towarow, ich ilosci i wartosci,")
					print(" -summ    wypisanie ogolnych danych o towarach (tzn.: liczby produktow w spisie, ilosci wszystkich towarow i ich wartosci),")		
					print(" -print   wypisanie danych o towarach do pliku - Summary_py.txt. W pliku znajda sie informacje o produktach: dla kazdego podana jego ilosc oraz cena, a takze podsumowanie, ktore zawiera liczbe wszystkich produktow w spisie, ilosci wszystkich towarow w spisie oraz ich laczna wartosc.\n\n")

					print("Przykladowy plik, na ktorym operuje skrypt: magazyn.csv.\nPrzyklad wywolania skryptu: python3 sIG_zal.py magazyn.csv -h -summ\n");
					h+=1

		elif isfile(argv[pom]):
			if f==0:
				file=argv[pom]
			f+=1

	for pom in range (1, len(argv)):
		if f==1:
			if argv[pom] == '-add':
				addProduct(file)
			elif argv[pom] == '-rem':
				removeProduct(file)
			elif argv[pom] == '-change':
				changeProduct(file)
			elif argv[pom] == '-show':
				showProducts(file)
			elif argv[pom] == '-summ':
				summary(file)
			elif argv[pom] == '-print':
				printFile(file)
			elif argv[pom] == '-h' or argv[pom] == '-help':
				h+=1
			elif isfile(argv[pom]):
				pass
			else:
				print("\nNie znam takiej opcji -'" +argv[pom] +"'. Zajrzyj do pomocy :)\n")
		
if f>1:
	print(kolor.RED +"Chyba cos jest nie tak - podales za duzo plikow. Sprawdz pomoc i wroc do mnie :)" +kolor.RESET)
if f==0 and h==0:
	print (kolor.RED +"Nie podano pliku do analizy :(" +kolor.RESET);

if f==1 and len(argv)==2:
	print("PRODUKTY z magazynu - " +file)
	print("Nazwa produktu, ilosc, cena\n===========================")
	with open(file) as csv_file:
		csv_reader = csv.reader(csv_file, delimiter=',')
		line_count = 0
		lil=0
		for row in csv_reader:
			print (row[0], row[1], row[2], row[3])
			line_count += 1
			lil+=int(row[1])
	print("\nW magazynie jest " +str(line_count) +" towarow.")
	print("Lacznie na magazynie jest: " +str(lil) +" sztuk wszystkich produktow.")

File: perl/projekt_spis_z_natury/sIG_zal.pl
#!/usr/bin/perl
#Izabela Górlińska grupa 1
#skrypt zaliczeniowy perl
#chmod +x sIG_zal.pl

use Cwd qw( abs_path );
use File::Basename qw( dirname );
BEGIN {
        push @INC,dirname(abs_path($0));;
}
use Istnieje;
use strict;
use warnings;
use Scalar::Util qw(looks_like_number);
use Term::ANSIColor qw(:constants);


sub addProduct{ 
	my($spis)=@_; #(spis)
	print("\nDodajesz produkt do spisu - $spis. Podaj dane produktu.");
	print "\nNazwa: ";
	my $nazwa=<STDIN>;
	chomp $nazwa;
	print "Ilosc: ";
	my $ilosc=<STDIN>;
	my $i=0;
	while ($i<3) {
		unless($ilosc =~ /^[0-9]/) {
			if($i==0) {
				print "To nie liczba sztuk! Podaj poprawnie: ";
				$ilosc=<STDIN>;
				chomp $nazwa;
			}
			elsif($i==2) {
				print RED"Podano nieprawidlowa wartosc! Nie udalo sie dodac produktu :(\n", RESET;
				return;
			}
			else {
				print "Sprobuj jeszcze raz: ";
				$ilosc=<STDIN>;
				chomp $nazwa;
			}
			$i++;
		}else {
			$i=3;
		}
	}
	print "Cena: ";
	my $cena=<STDIN>;
	unless(looks_like_number($cena)) {
		print "Podaj poprawna cene: ";
		$cena=<STDIN>;
		unless(looks_like_number($cena)) {
			print "Podales niepoprawna cene. Sprobuj jeszcze raz: ";
			$cena=<STDIN>;
		}
		unless(looks_like_number($cena)) {
			print RED"Wprowadzona cena nie jest poprawna. Przerwano dodawanie produktu\n", RESET;
			return
		}
	}
	my $suma=$ilosc*$cena;
	open(PLIK, '>>', $spis) or die "Nie można utworzyć pliku: $!";
	printf PLIK "%s,%d,%.2f,%.2f\n",$nazwa, $ilosc, $cena, $suma;
	close PLIK;
	print GREEN"Pomyslnie dodano produkt do spisu :)\n", RESET;
}


sub removeProduct{
	my($spis)=@_; #(spis)
	print("\nUsuwasz produkt ze spisu - $spis.\nPodaj nazwe produktu, ktory chcesz usunac: ");
	my $prod=<STDIN>;
	chomp $prod;
	my @list=();
	my $usun=0;
	open(PLIK, '<', $spis) or die "Nie można utworzyć pliku: $!";
	while (my $line = <PLIK>) {
		chomp $line;
		my @slowawl = split(',', $line);
		if($slowawl[0] eq $prod){
			print "$prod jest w linii $line";
			print("\nCzy na pewno chcesz usunac $prod? (tak/nie) ");
			my $dec=<STDIN>;
			chomp $dec;
			if($dec eq 'tak'){
				$usun=1;	
			}else{
				print "Ok. Nie bedziemy usuwac\n";
				return;
			}
		}else{
			push(@list, $line);
		}
	}
	close PLIK;
	if($usun==1){
		open(PLIK, '>', $spis) or die "Nie można utworzyć pliku: $!";
		foreach my $p(@list){
			printf PLIK "$p\n";
		}
		close PLIK;
		print GREEN"\nProdukt zostal usuniety.\n", RESET;
		showProducts($spis);
	}else{
		print RED"Nie ma produktu o takiej nazwie :(\n", RESET;
	}
}



sub changeProduct{
	my($spis)=@_; #(spis)
	my $ex;
	my $prod;
	my $akcja;
	my $dec;
	print "\nZmieniasz dane produktu ze spisu - $spis";
	print "\nWybierz co chcesz zmienic:\n 1)nazwa\n 2)ilosc\n 3)cena\nTwoj wybor: ";
	$akcja=<STDIN>;
	chomp $akcja;
	if($akcja eq '1'){
		print "Zmieniasz nazwe.\nPodaj nazwe produktu, ktory chcesz edytowac: ";
		$prod=<STDIN>;
		chomp $prod;
		$ex=Istnieje::istnieje($spis, $prod, 0);
		unless($ex==1){
			print "\nPodaj nazwe produktu jeszcze raz: ";
			$prod=<STDIN>;
			chomp $prod;
			$ex=Istnieje::istnieje($spis, $prod, 1);
		}
		if($ex==1){
			my @list=();
			my @proded;	
			open(PLIK, '<', $spis) or die "Nie można otworzyć pliku: $!";
			while (my $line = <PLIK>) {
				chomp $line;
				my @slowawl = split(',', $line);
				if($slowawl[0] eq $prod){
					@proded=@slowawl;
					print "Podaj nowa nazwe: ";
					my $nowa=<STDIN>;
					chomp $nowa;
					$proded[0]=$nowa;
					my $pr=join( ",",$proded[0],$proded[1],$proded[2],$proded[3]);
					push(@list, $pr);
					print("Czy na pewno chcesz zmienic nazwe $prod na $nowa? (tak/nie)");
					$dec=<STDIN>;
					chomp $dec;
				}else{
					push(@list, $line);
				}
			}
			close PLIK;
			
			if($dec eq 'tak'){
				open(PLIK, '>', $spis) or die "Nie można otworzyć pliku: $!";
				foreach my $p(@list){
					printf PLIK "$p\n";
				}
				close PLIK;
				print GREEN"Edycja zakonczona sukcesem :)", RESET;
			}else{
				print RED"\nEdycja wstrzymana", RESET;
				return;
			}
		}
	}
	elsif($akcja eq '2'){
		print "Zmieniasz ilosc.\nPodaj nazwe produktu, ktory chcesz edytowac: ";
		my $prod=<STDIN>;
		chomp $prod;
		$ex=Istnieje::istnieje($spis, $prod, 0);
		unless($ex==1){
			print "\nPodaj nazwe produktu jeszcze raz: ";
			$prod=<STDIN>;
			chomp $prod;
			$ex=Istnieje::istnieje($spis, $prod, 1);
		}
		if($ex==1){
			my @list=();
			my @proded;	
			open(PLIK, '<', $spis) or die "Nie można otworzyć pliku: $!";
			while (my $line = <PLIK>) {
				chomp $line;
				my @slowawl = split(',', $line);
				if($slowawl[0] eq $prod){
					@proded=@slowawl;
					print "Podaj nowa ilosc: ";
					my $nowa=<STDIN>;
					chomp $nowa;
					unless($nowa =~ /^[0-9]/) {
						print "Podaj poprawna ilosc: ";
						$nowa=<STDIN>;
						unless($nowa =~ /^[0-9]/) {
							print "Podales niepoprawna ilosc. Sprobuj jeszcze raz: ";
							$nowa=<STDIN>;
						}
						unless($nowa =~ /^[0-9]/) {
							print "Wprowadzona ilosc nie jest poprawna. Przerwano zmienianie ilosci produktu\n";
							return
						}
					}
					$proded[1]=$nowa;
					$proded[3]=$proded[1]*$proded[2];
					$proded[3]=int($proded[3]*100)/100;
					my $pr=join( ",",$proded[0],$proded[1],$proded[2],$proded[3]);
					push(@list, $pr);
					print("Czy na pewno chcesz zmienic ilosc $prod? (tak/nie)");
					$dec=<STDIN>;
					chomp $dec;
				}else{
					push(@list, $line);
				}
			}
			close PLIK;
			
			if($dec eq 'tak'){
				open(PLIK, '>', $spis) or die "Nie można otworzyć pliku: $!";
				foreach my $p(@list){
					printf PLIK "$p\n";
				}
				close PLIK;
				print GREEN"Edycja zakonczona sukcesem :)", RESET;
			}else{
				print RED"\nEdycja wstrzymana", RESET;
				return;
			}
		}
	}
	elsif($akcja eq '3'){
		print "Zmieniasz cene.\nPodaj nazwe produktu, ktory chcesz edytowac: ";
		my $prod=<STDIN>;
		chomp $prod;
		$ex=Istnieje::istnieje($spis, $prod, 0);
		unless($ex==1){
			print "\nPodaj nazwe produktu jeszcze raz: ";
			$prod=<STDIN>;
			chomp $prod;
			$ex=Istnieje::istnieje($spis, $prod, 1);
		}
		if($ex==1){
			my @list=();
			my @proded;	
			open(PLIK, '<', $spis) or die "Nie można otworzyć pliku: $!";
			while (my $line = <PLIK>) {
				chomp $line;
				my @slowawl = split(',', $line);
				if($slowawl[0] eq $prod){
					@proded=@slowawl;
					print "Podaj nowa cene: ";
					my $nowa=<STDIN>;
					chomp $nowa;
					unless(looks_like_number($nowa)) {
						print "Podaj poprawna cene: ";
						$nowa=<STDIN>;
						unless(looks_like_number($nowa)) {
							print "Podales niepoprawna cene. Sprobuj jeszcze raz: ";
							$nowa=<STDIN>;
						}
						unless(looks_like_number($nowa)) {
							print "Wprowadzona cena nie jest poprawna. Przerwano zmienianie ceny produktu\n";
							return
						}
					}
					$proded[2]=$nowa;
					$proded[2]=int($proded[2]*100)/100;
					$proded[3]=$proded[1]*$proded[2];
					$proded[3]=int($proded[3]*100)/100;
					my $pr=join( ",",$proded[0],$proded[1],$proded[2],$proded[3]);
					push(@list, $pr);
					print("Czy na pewno chcesz zmienic cene $prod? (tak/nie)");
					$dec=<STDIN>;
					chomp $dec;
				}else{
					push(@list, $line);
				}
			}
			close PLIK;
			
			if($dec eq 'tak'){
				open(PLIK, '>', $spis) or die "Nie można otworzyć pliku: $!";
				foreach my $p(@list){
					printf PLIK "$p\n";
				}
				close PLIK;
				print GREEN"Edycja zakonczona sukcesem :)", RESET;
			}else{
				print RED"\nEdycja wstrzymana", RESET;
				return;
			}
		}
	}else{
		print RED"Nie wybrano numeru akcji. Uruchom skrypt ponownie jesli chcesz edytowac", RESET;
	}
}


sub showProducts{
	my($spis)=@_; #(spis)
	print("\nWyswietlam spis - $spis\n");
	open(PLIK, '<', $spis) or die "Nie można utworzyć pliku: $!";
	while(my $line = <PLIK>){
		chomp $line;
		my @slowawl = split(',', $line);
		print "$slowawl[0] $slowawl[1] $slowawl[2] $slowawl[3]\n";
	}
	close PLIK;
}


sub summary{
	my($spis)=@_;
	print "\nWyswietlam dane ogolne dla spisu - $spis";
	my $lprod=0;
	my $lil=0;
	my $lsum=0;
	open(PLIK, '<', $spis) or die "Nie można otworzyć pliku: $!";
	while(my $line = <PLIK>){
		chomp $line;
		my @slowawl = split(',', $line);
		$lprod++;
		$lil+=int($slowawl[1]);
		$lsum+=$slowawl[3];
		$lsum=int($lsum*100)/100;
	}
	close PLIK;
	print "\nLiczba produktow w spisie: $lprod\nLacznie na magazynie jest: $lil sztuk wszystkich towarow\nWartosc wszystkich produktow wynosi: $lsum zl\n";
}


sub printFile{
	my($spis)=@_;
	my $lprod=0;
	my $lil=0;
	my $lsum=0;
	open(PLIK1, '>', 'Summary_pl.txt') or die "Nie można utworzyć pliku: $!";
	printf PLIK1 "Podsumowanie spisu towarow z magazynu\n\nPRODUKTY\nNazwa produktu, ilosc, cena\n===========================\n";
	open(PLIK, '<', $spis) or die "Nie można otworzyć pliku: $!";
	while(my $line = <PLIK>){
		chomp $line;
		my @slowawl = split(',', $line);
		printf PLIK1 "$slowawl[0] $slowawl[1] $slowawl[2] $slowawl[3]\n";
		$lprod++;
		$lil+=int($slowawl[1]);
		$lsum+=$slowawl[3];
		$lsum=int($lsum*100)/100;
	}
	close PLIK;
	printf PLIK1 "\nPODSUMOWANIE\nLiczba produktow w spisie: $lprod\nLacznie na magazynie jest: $lil sztuk wszystkich towarow\nWartosc wszystkich produktow wynosi: $lsum zl";
	close PLIK1;
}


#koniec sub
my $file;
my $pom=0;
my $f=0;
my $h=0;

unless (defined($ARGV[0])) {
	print RED"Nie podano pliku do analizy\nJesli chcesz dowiedziec sie co ja robie wybierz opcje '-help' :)\n", RESET;
	exit 1;
}else {
	for $pom (@ARGV) {
		if (( -f $pom)) {
			if (( $f==0)){
				$file=$pom;
			}
			$f++;
		}
		if (( $pom eq '-h' ) || ( $pom eq '-help')) {
			if(( $h==0)){
				print "Wywolales skrypt z opcja help.\n";
				print "Autor: Izabela Gorlinska\n\n";
				print "Ten program jako argument przyjmuje plik z danymi(plik csv) - spisem stanu towarow na magazynie i na nim operuje.\nProgram akceptuje tylko 1 plik i na nim pracuje.\nPo wywolaniu konkretnej opcji wykonuje odpowiednia dla niej akcje.\nSkrypt mozna uruchomic z 1 lub wieloma opcjami na raz.\n";
				print "\nSkrypt wywolany tylko z pierwszym argumenten (plikiem na ktorym operuje) wypisuje wartosci wszystkich towarow w magazynie oraz liczbę towarów i ich laczna ilosc.\nMozliwe opcje wywolania skryptu:\n";
				print " -add     dodanie produktu do magazynu,\n";
				print " -rem     usuniecie produktu z magazynu,\n";
				print " -change  zmiana nazwy,ilosci lub ceny produktu,\n";
				print " -show    wypisanie towarow, ich ilosci i wartosci,\n";
				print " -summ    wypisanie ogolnych danych o towarach (tzn.: liczby produktow w spisie, ilosci wszystkich towarow i ich wartosci),\n";		
				print " -print   wypisanie danych o towarach do pliku- Summary_pl.txt. W pliku znajda sie informacje o produktach: dla kazdego podana jego ilosc oraz cena, a takze podsumowanie, ktore zawiera liczbe wszystkich produktow w spisie, ilosci wszystkich towarow w spisie oraz ich laczna wartosc.\n\n";
				print "\nPrzykladowy plik, na ktorym operuje skrypt: magazyn.csv.\nPrzyklad wywolania skryptu: ./sIG_zal.pl magazyn.csv -summ -help\n\n";
				$h++;
			}
		}
	}
	for $pom (@ARGV) {
		if($f==1){
			if (( $pom eq '-add' ) ) {
				addProduct($file);
			}
			elsif (( $pom eq '-rem' )) {
				removeProduct($file);
			}
			elsif (( $pom eq '-change' )) {
				changeProduct($file);
			}
			elsif (( $pom eq '-show' )) {
				showProducts($file);
			}
			elsif (( $pom eq '-summ' )) {
				summary($file);
			}
			elsif (( $pom eq '-print' )) {
				printFile($file);
			}
			elsif (( $pom eq '-h' ) || ( $pom eq '-help')) {
				$h++;
			}
			elsif (( -f $pom)) {
			}
			else{
				print "\nNie znam takiej opcji -'$pom'. Zajrzyj do pomocy :)\n";
			}
		}
	}
}
if (( $f==0 ) && ( $h==0)){
	print RED"Nie podano pliku do analizy ;(", RESET;
}
elsif (( $f>1)){
	print RED"Podales za duzo plikow. Zgubilem sie :(\nPrzeczytaj pomoc i sprobuj jeszcze raz!", RESET;
}

if (( $f==1 ) && ( $#ARGV==0 )){
	print "PRODUKTY z magazynu - $file\n\n";
	print "Nazwa produktu, ilosc, cena\n===========================\n";
	my $lprod=0;
	my $lil=0;
	open(PLIK, '<', $file) or die "Nie można otworzyć pliku: $!";
	while(my $line = <PLIK>){
		chomp $line;
		my @slowawl = split(',', $line);
		print "$slowawl[0] $slowawl[1] $slowawl[2] $slowawl[3]\n";
		$lprod++;
		$lil+=int($slowawl[1]);
	}
	close PLIK;
	print "\nW magazynie jest $lprod produktow.\nLacznie na magazynie jest $lil sztuk wszystkich produktow."
}
print "\n"
